// @sentry/utils/build/cjs/index.js
Object.defineProperty(exports, '__esModule', { value: true });

const core = require('@sentry/core');



exports.BAGGAGE_HEADER_NAME = core.BAGGAGE_HEADER_NAME;
exports.CONSOLE_LEVELS = core.CONSOLE_LEVELS;
exports.DEFAULT_RETRY_AFTER = core.DEFAULT_RETRY_AFTER;
exports.DEFAULT_USER_INCLUDES = core.DEFAULT_USER_INCLUDES;
exports.GLOBAL_OBJ = core.GLOBAL_OBJ;
exports.LRUMap = core.LRUMap;
exports.MAX_BAGGAGE_STRING_LENGTH = core.MAX_BAGGAGE_STRING_LENGTH;
exports.SDK_VERSION = core.SDK_VERSION;
exports.SENTRY_BAGGAGE_KEY_PREFIX = core.SENTRY_BAGGAGE_KEY_PREFIX;
exports.SENTRY_BAGGAGE_KEY_PREFIX_REGEX = core.SENTRY_BAGGAGE_KEY_PREFIX_REGEX;
exports.SentryError = core.SentryError;
exports.SyncPromise = core.SyncPromise;
exports.TRACEPARENT_REGEXP = core.TRACEPARENT_REGEXP;
exports.UNKNOWN_FUNCTION = core.UNKNOWN_FUNCTION;
exports._asyncNullishCoalesce = core._asyncNullishCoalesce;
exports._asyncOptionalChain = core._asyncOptionalChain;
exports._asyncOptionalChainDelete = core._asyncOptionalChainDelete;
exports._browserPerformanceTimeOriginMode = core._browserPerformanceTimeOriginMode;
exports._nullishCoalesce = core._nullishCoalesce;
exports._optionalChain = core._optionalChain;
exports._optionalChainDelete = core._optionalChainDelete;
exports.addConsoleInstrumentationHandler = core.addConsoleInstrumentationHandler;
exports.addContextToFrame = core.addContextToFrame;
exports.addExceptionMechanism = core.addExceptionMechanism;
exports.addExceptionTypeValue = core.addExceptionTypeValue;
exports.addFetchEndInstrumentationHandler = core.addFetchEndInstrumentationHandler;
exports.addFetchInstrumentationHandler = core.addFetchInstrumentationHandler;
exports.addGlobalErrorInstrumentationHandler = core.addGlobalErrorInstrumentationHandler;
exports.addGlobalUnhandledRejectionInstrumentationHandler = core.addGlobalUnhandledRejectionInstrumentationHandler;
exports.addHandler = core.addHandler;
exports.addItemToEnvelope = core.addItemToEnvelope;
exports.addNonEnumerableProperty = core.addNonEnumerableProperty;
exports.addNormalizedRequestDataToEvent = core.addNormalizedRequestDataToEvent;
exports.addRequestDataToEvent = core.addRequestDataToEvent;
exports.applyAggregateErrorsToEvent = core.applyAggregateErrorsToEvent;
exports.arrayify = core.arrayify;
exports.baggageHeaderToDynamicSamplingContext = core.baggageHeaderToDynamicSamplingContext;
exports.basename = core.basename;
exports.browserPerformanceTimeOrigin = core.browserPerformanceTimeOrigin;
exports.callFrameToStackFrame = core.callFrameToStackFrame;
exports.checkOrSetAlreadyCaught = core.checkOrSetAlreadyCaught;
exports.consoleSandbox = core.consoleSandbox;
exports.convertToPlainObject = core.convertToPlainObject;
exports.createAttachmentEnvelopeItem = core.createAttachmentEnvelopeItem;
exports.createClientReportEnvelope = core.createClientReportEnvelope;
exports.createEnvelope = core.createEnvelope;
exports.createEventEnvelopeHeaders = core.createEventEnvelopeHeaders;
exports.createSpanEnvelopeItem = core.createSpanEnvelopeItem;
exports.createStackParser = core.createStackParser;
exports.dateTimestampInSeconds = core.dateTimestampInSeconds;
exports.dirname = core.dirname;
exports.disabledUntil = core.disabledUntil;
exports.dropUndefinedKeys = core.dropUndefinedKeys;
exports.dsnFromString = core.dsnFromString;
exports.dsnToString = core.dsnToString;
exports.dynamicRequire = core.dynamicRequire;
exports.dynamicSamplingContextToSentryBaggageHeader = core.dynamicSamplingContextToSentryBaggageHeader;
exports.envelopeContainsItemType = core.envelopeContainsItemType;
exports.envelopeItemTypeToDataCategory = core.envelopeItemTypeToDataCategory;
exports.escapeStringForRegex = core.escapeStringForRegex;
exports.eventFromMessage = core.eventFromMessage;
exports.eventFromUnknownInput = core.eventFromUnknownInput;
exports.exceptionFromError = core.exceptionFromError;
exports.extractExceptionKeysForMessage = core.extractExceptionKeysForMessage;
exports.extractPathForTransaction = core.extractPathForTransaction;
exports.extractRequestData = core.extractRequestData;
exports.extractTraceparentData = core.extractTraceparentData;
exports.filenameIsInApp = core.filenameIsInApp;
exports.fill = core.fill;
exports.flatten = core.flatten;
exports.forEachEnvelopeItem = core.forEachEnvelopeItem;
exports.generatePropagationContext = core.generatePropagationContext;
exports.generateSentryTraceHeader = core.generateSentryTraceHeader;
exports.getBreadcrumbLogLevelFromHttpStatusCode = core.getBreadcrumbLogLevelFromHttpStatusCode;
exports.getComponentName = core.getComponentName;
exports.getDebugImagesForResources = core.getDebugImagesForResources;
exports.getDomElement = core.getDomElement;
exports.getEventDescription = core.getEventDescription;
exports.getFilenameToDebugIdMap = core.getFilenameToDebugIdMap;
exports.getFramesFromEvent = core.getFramesFromEvent;
exports.getFunctionName = core.getFunctionName;
exports.getGlobalSingleton = core.getGlobalSingleton;
exports.getLocationHref = core.getLocationHref;
exports.getNumberOfUrlSegments = core.getNumberOfUrlSegments;
exports.getOriginalFunction = core.getOriginalFunction;
exports.getSDKSource = core.getSDKSource;
exports.getSanitizedUrlString = core.getSanitizedUrlString;
exports.getSdkMetadataForEnvelopeHeader = core.getSdkMetadataForEnvelopeHeader;
exports.htmlTreeAsString = core.htmlTreeAsString;
exports.isAbsolute = core.isAbsolute;
exports.isBrowser = core.isBrowser;
exports.isBrowserBundle = core.isBrowserBundle;
exports.isDOMError = core.isDOMError;
exports.isDOMException = core.isDOMException;
exports.isElement = core.isElement;
exports.isError = core.isError;
exports.isErrorEvent = core.isErrorEvent;
exports.isEvent = core.isEvent;
exports.isInstanceOf = core.isInstanceOf;
exports.isMatchingPattern = core.isMatchingPattern;
exports.isNativeFunction = core.isNativeFunction;
exports.isNodeEnv = core.isNodeEnv;
exports.isParameterizedString = core.isParameterizedString;
exports.isPlainObject = core.isPlainObject;
exports.isPrimitive = core.isPrimitive;
exports.isRateLimited = core.isRateLimited;
exports.isRegExp = core.isRegExp;
exports.isString = core.isString;
exports.isSyntheticEvent = core.isSyntheticEvent;
exports.isThenable = core.isThenable;
exports.isVueViewModel = core.isVueViewModel;
exports.join = core.join;
exports.loadModule = core.loadModule;
exports.logger = core.logger;
exports.makeDsn = core.makeDsn;
exports.makeFifoCache = core.makeFifoCache;
exports.makePromiseBuffer = core.makePromiseBuffer;
exports.markFunctionWrapped = core.markFunctionWrapped;
exports.maybeInstrument = core.maybeInstrument;
exports.memoBuilder = core.memoBuilder;
exports.node = core.node;
exports.nodeStackLineParser = core.nodeStackLineParser;
exports.normalize = core.normalize;
exports.normalizePath = core.normalizePath;
exports.normalizeToSize = core.normalizeToSize;
exports.normalizeUrlToBase = core.normalizeUrlToBase;
exports.objectify = core.objectify;
exports.originalConsoleMethods = core.originalConsoleMethods;
exports.parseBaggageHeader = core.parseBaggageHeader;
exports.parseEnvelope = core.parseEnvelope;
exports.parseRetryAfterHeader = core.parseRetryAfterHeader;
exports.parseSemver = core.parseSemver;
exports.parseStackFrames = core.parseStackFrames;
exports.parseUrl = core.parseUrl;
exports.propagationContextFromHeaders = core.propagationContextFromHeaders;
exports.rejectedSyncPromise = core.rejectedSyncPromise;
exports.relative = core.relative;
exports.resetInstrumentationHandlers = core.resetInstrumentationHandlers;
exports.resolve = core.resolve;
exports.resolvedSyncPromise = core.resolvedSyncPromise;
exports.safeJoin = core.safeJoin;
exports.serializeEnvelope = core.serializeEnvelope;
exports.severityLevelFromString = core.severityLevelFromString;
exports.snipLine = core.snipLine;
exports.stackParserFromStackParserOptions = core.stackParserFromStackParserOptions;
exports.stringMatchesSomePattern = core.stringMatchesSomePattern;
exports.stripSentryFramesAndReverse = core.stripSentryFramesAndReverse;
exports.stripUrlQueryAndFragment = core.stripUrlQueryAndFragment;
exports.supportsDOMError = core.supportsDOMError;
exports.supportsDOMException = core.supportsDOMException;
exports.supportsErrorEvent = core.supportsErrorEvent;
exports.supportsFetch = core.supportsFetch;
exports.supportsHistory = core.supportsHistory;
exports.supportsNativeFetch = core.supportsNativeFetch;
exports.supportsReferrerPolicy = core.supportsReferrerPolicy;
exports.supportsReportingObserver = core.supportsReportingObserver;
exports.timestampInSeconds = core.timestampInSeconds;
exports.triggerHandlers = core.triggerHandlers;
exports.truncate = core.truncate;
exports.updateRateLimits = core.updateRateLimits;
exports.urlEncode = core.urlEncode;
exports.uuid4 = core.uuid4;
exports.validSeverityLevels = core.validSeverityLevels;
exports.vercelWaitUntil = core.vercelWaitUntil;
exports.watchdogTimer = core.watchdogTimer;
exports.winterCGHeadersToDict = core.winterCGHeadersToDict;
exports.winterCGRequestToRequestData = core.winterCGRequestToRequestData;
